src/Physics/BeamHNL/HNLDecayer.h

```diff
@@ -0,0 +1,117 @@
+//____________________________________________________________________________
+/*!
+
+\class    genie::hnl::Decayer
+
+\brief    Heavy Neutral Lepton final-state product generator
+
+\author   Costas Andreopoulos <constantinos.andreopoulos \at cern.ch>
+          University of Liverpool & STFC Rutherford Appleton Laboratory
+	  John Plows <komninos-john.plows \at physics.ox.ac.uk>
+
+\created  February 10, 2020
+
+\cpright  Copyright (c) 2003-2022, The GENIE Collaboration
+          For the full text of the license visit http://copyright.genie-mc.org        
+*/
+//____________________________________________________________________________
+
+#ifndef _HNL_DECAYER_H_
+#define _HNL_DECAYER_H_
+
+#include <cassert>
+
+#include <TGenPhaseSpace.h>
+#include <TH3.h>
+
+#include "Framework/EventGen/EventRecordVisitorI.h"
+#include "Framework/Numerical/RandomGen.h"
+#include "Framework/ParticleData/PDGCodes.h"
+
+#include "Tools/Flux/GNuMIFlux.h"
+
+#include "Physics/BeamHNL/HNLDecayMode.h"
+#include "Physics/BeamHNL/SimpleHNL.h"
+
+namespace genie {
+
+namespace hnl {
+
+class Decayer: public EventRecordVisitorI {
+
+public:
+  Decayer();
+  Decayer(string config);
+ ~Decayer();
+
+  // implement the EventRecordVisitorI interface
+  void ProcessEventRecord (GHepRecord * event) const;
+
+  // overload the Algorithm::Configure() methods to load private data
+  // members from configuration options
+  void Configure(const Registry & config);
+  void Configure(string config);
+
+private:
+
+   void LoadConfig            (void);
+   void SetEnvVariable        (const char * var, double value) const;
```

Comment:
This looks like something that breaks interfaces, and communicates with other modules in a non-standard way, be sting environmental variables.  Why do we need this?

Reply:
Env variables manipulation has been removed. Calls to necessary information are handled in dedicated interfaces now.